src/typing/type_inference_js.ml

```diff
@@ -797,6 +796,21 @@ let rec convert cx map = Ast.Type.(function
             ((snd name).Ast.Identifier.name) :: rev_pnames
         | None -> rev_tlist, rev_pnames
       ) in
+    let ret = convert cx map returnType in
+    let ret =
+      if generator then
+        let reason = mk_reason "generator type" loc in
+        let yield = Flow_js.mk_tvar cx (prefix_reason "yield of " reason) in
+        let return = Flow_js.mk_tvar cx (prefix_reason "return of " reason) in
+        let next = Flow_js.mk_tvar cx (prefix_reason "next of " reason) in
+        let gt = Flow_js.get_builtin_typeapp
+          cx
+          reason
+          "Generator"
+          [yield; return; next] in
+        Flow_js.flow cx (ret, gt);
+        gt
+      else ret in
```

Comment:
Above is where the "hijacking" which I mentioned in the PR description takes place.